tests/Microsoft.Identity.Web.Test/WebAppExtensionsTests.cs

```diff
@@ -1,32 +1,493 @@
+
+            // Assert configure options actions added correctly
+            var configuredOidcOptions = provider.GetServices<IConfigureOptions<OpenIdConnectOptions>>().Cast<ConfigureNamedOptions<OpenIdConnectOptions>>();
+            var configuredMsOptions = provider.GetServices<IConfigureOptions<MicrosoftIdentityOptions>>().Cast<ConfigureNamedOptions<MicrosoftIdentityOptions>>();
+
+            Assert.Contains(configuredOidcOptions, o => o.Action == _configureOidcOptions);
+            Assert.Contains(configuredMsOptions, o => o.Action == _configureMsOptions);
+
+            AddSignIn_TestCommon(services, provider);
+        }
+
+        private void AddSignIn_TestCommon(IServiceCollection services, ServiceProvider provider)
+        {
+            // Assert correct services added           
+            Assert.Contains(services, s => s.ServiceType == typeof(IConfigureOptions<OpenIdConnectOptions>));
+            Assert.Contains(services, s => s.ServiceType == typeof(IConfigureOptions<MicrosoftIdentityOptions>));
+            Assert.Contains(services, s => s.ServiceType == typeof(IOpenIdConnectMiddlewareDiagnostics));
+            Assert.Equal(ServiceLifetime.Singleton, services.First(s => s.ServiceType == typeof(IOpenIdConnectMiddlewareDiagnostics)).Lifetime);
+            Assert.Contains(services, s => s.ServiceType == typeof(IPostConfigureOptions<CookieAuthenticationOptions>));
+
+            // Assert OIDC options added correctly
+            var configuredOidcOptions = provider.GetService<IConfigureOptions<OpenIdConnectOptions>>() as ConfigureNamedOptions<OpenIdConnectOptions>;
+
+            Assert.Equal(_oidcScheme, configuredOidcOptions.Name);
+
+            // Assert properties set
+            var oidcOptions = provider.GetRequiredService<IOptionsFactory<OpenIdConnectOptions>>().Create(_oidcScheme);
+
+            Assert.Equal(_cookieScheme, oidcOptions.SignInScheme);
+            Assert.NotNull(oidcOptions.Authority);
+            Assert.NotNull(oidcOptions.TokenValidationParameters.IssuerValidator);
+            Assert.Equal(ClaimConstants.PreferredUserName, oidcOptions.TokenValidationParameters.NameClaimType);
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task AddSignIn_WithConfigName_OnRedirectToIdentityProviderEvent(bool useServiceCollectionExtension)
+        {
+            var configMock = Substitute.For<IConfiguration>();
+            configMock.Configure().GetSection(_configSectionName).Returns(_configSection);
+
+            var redirectFunc = Substitute.For<Func<RedirectContext, Task>>();
+            var services = new ServiceCollection()
+                .Configure<OpenIdConnectOptions>(_oidcScheme, (options) => {
+                    options.Events ??= new OpenIdConnectEvents();
+                    options.Events.OnRedirectToIdentityProvider += redirectFunc;
+                });
+            services.AddDataProtection();
```

Comment:
what does `.AddDataProtection()` do?

Reply:
I haven't looked exactly at what it does but the data protection services are required when creating an instance of OIDC options.

Reply:
I think that this is the DPAPI?

Reply:
Should we have it in the product code as well? @pmaytak ?

Reply:
In prod code [AddAuthentication](https://github.com/dotnet/aspnetcore/blob/c9ce0664ea2d9ff842af8f6242d1609c93e8b683/src/Security/Authentication/Core/src/AuthenticationServiceCollectionExtensions.cs#L16) method adds the data protection services.